apps/tsilhqotin-language-hub/src/Components/Pages/SongDetail/SongDetail.tsx

```diff
@@ -1,16 +1,37 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import SongData, { Song } from '../Song/Song';
 import './SongDetail.module.css';
 
 /* eslint-disable-next-line */
 export interface SongViewModel {}
 
-export function SongDetail(_props: SongViewModel) {
-    return (
-        <div>
-            {/*
-            <Songs songData={componentState.songData}></Songs>
-            */}
-        </div>
-    );
+type ComponentState = {
+    songData: null | SongData;
+};
+
+export function SongDetail(props: SongViewModel) {
+    const [componentState, setComponentState] = useState<ComponentState>({
+        songData: null,
+    });
+
+    const { id } = useParams();
+
+    useEffect(() => {
+        setComponentState({ songData: null });
+
+        const apiUrl = `http://localhost:3131/api/resources/songs/${id}`;
+        fetch(apiUrl, { mode: 'cors' })
+            .then((res) => res.json())
+            .then((song) => {
+                setComponentState({ songData: song.data });
+            })
+            .catch((rej) => console.log(rej));
+    }, [setComponentState]);
+
+    if (!componentState.songData) return <div>LOADING!</div>;
```

Comment:
Have you thought about what to do if the http request times out or fails? An error message and timely redirect would be better than indefinitely loading, but let's not worry about it just yet.